torch/_decomp/decompositions.py

```diff
@@ -1963,6 +1963,12 @@ def uniform(
     )
 
 
+@register_decomposition(aten.uniform_)
+def uniform_(self, low=0, high=1, generator=None):
+    assert generator is None
```

Comment:
Looks like the assert fails on some crossref mode CIs. 

@jansel, may I know why the `generator` arg in decomp is just ignored here:
https://github.com/pytorch/pytorch/blob/f21997099087c6db85e119cd5ef97e2d8a134a85/torch/_decomp/decompositions.py#L1522-L1527

And the context behind
https://github.com/pytorch/pytorch/blob/f21997099087c6db85e119cd5ef97e2d8a134a85/torch/_prims/__init__.py#L2676-L2677
I wonder if it's OK to ignore the `generator` arg from the decomp side if the backend can still get the seed from the default generator (if the default generator is global).

Reply:
I didn't write that code.  Though that decomp seems wrong if the user provides a non-default generator.  We should fix that.

Reply:
Yeah we should assert that generator is default or none, otherwise decomp is wrong. Currently inductor is not set up to respect non-default generator. We probably need to skip the test on crossref builds.

Reply:
Thanks @jansel, @ngimel, understood.  decomp should respect the function args, so better not just ignore `generator` arg. 

Done, https://github.com/pytorch/pytorch/pull/91833.